addons/account/tests/test_account_account.py

```diff
@@ -623,6 +623,71 @@ def test_generate_account_suggestions(self):
         )
         self.assertFalse(account.id in results_2, "Deprecated account should NOT appear in account suggestions")
 
+    def test_placeholder_code(self):
+        """ Test that the placeholder code is '{code_in_company} ({company})'
+            where `company` is the first company in the environment that is
+            in `account.company_ids`.
+
+            Check that `_field_to_sql` gives the same value.
+        """
+        def get_placeholder_code_via_sql(account):
+            account_query = account._as_query()
+            placeholder_code_sql = account_query.select(account._field_to_sql('account_account', 'placeholder_code', account_query))
+            placeholder_code = self.env.execute_query(placeholder_code_sql)[0][0]
+            return placeholder_code
+
+        branch = self.env['res.company'].create({
+            'name': 'Branch',
+            'parent_id': self.company_data['company'].id,
+        })
+        account = self.env['account.account'].create([{
+            'name': 'My account',
+            'company_ids': [Command.set(self.company_data['company'].ids)],
+            'code_mapping_ids': [
+                Command.create({'company_id': self.company_data['company'].id, 'code': '180001'}),
+            ]
+        }])
+
+        # Use `with_company` in order to force an order for `allowed_company_ids` in the context.
+        self.assertEqual(account.with_company(self.company_data['company']).placeholder_code, '180001')
+        self.assertEqual(get_placeholder_code_via_sql(account.with_company(self.company_data['company'])), '180001')
+
+        self.assertEqual(account.with_company(branch).placeholder_code, '180001')
+        self.assertEqual(get_placeholder_code_via_sql(account.with_company(branch)), '180001')
+
+        account_context_1 = account.with_context({
+            'allowed_company_ids': [self.company_data_2['company'].id, self.company_data['company'].id]
+        })
+        self.assertEqual(account_context_1.placeholder_code, '180001 (company_1_data)')
+        self.assertEqual(get_placeholder_code_via_sql(account_context_1), '180001 (company_1_data)')
+
+        account_context_2 = account.with_context({
+            'allowed_company_ids': [self.company_data_2['company'].id, branch.id]
+        })
+        self.assertEqual(account_context_2.placeholder_code, False)
+        self.assertEqual(get_placeholder_code_via_sql(account_context_2), None)
```

Comment:
I'm wondering why the need to check the `get_placeholder_code_via_sql`.

Wouldn't some call to `self.env['account.account'].search(['placeholder_code', '=', '180001 (company_1_data)'])` be a good enough validation of the behavior?
It would allow to push all those orm like details out of the tests.

But I may be missing something, I don't know that feature at all.

Reply:
Maybe I didn't understand it correctly but to me, it will end up calling this (directly or indirectly):
```py
    def _search_placeholder_code(self, operator, value):
        if operator != '=like':
            raise NotImplementedError
        query = Query(self.env, 'account_account')
        placeholder_code_sql = self.env['account.account']._field_to_sql('account_account', 'placeholder_code', query)
        query.add_where(SQL("%s LIKE %s", placeholder_code_sql, value))
        return [('id', 'in', query)]
```

Which itself call `_field_to_sql`, so, as far as I'm understanding, we just need to `self.env['account.account'].search(['placeholder_code', '=', '180001 (company_1_data)'])` to go through this piece of code.

Reply:
oh yes that's true, I had forgotten about this :smile:

Reply:
You can r- or force push :wink: